node/tests/swamp/swamp.go

```diff
@@ -63,9 +61,30 @@ func NewSwamp(t *testing.T) *Swamp {
 	swp.t.Cleanup(func() {
 		swp.stopAllNodes(ctx, swp.BridgeNodes, swp.LightNodes)
 	})
+
 	return swp
 }
 
+// TODO(@Bidon15): CoreClient(limitation)
+// Now, we are making an assumption that consensus mechanism is already tested out
+// so, we are not creating bridge nodes with each one containing its own core client
+// instead we are assigning all created BNs to 1 Core from the swamp
+
+// newTendermintCoreNode creates a new instance of Tendermint Core with a kvStore
+func newTendermintCoreNode(cfg *Config) (*tn.Node, error) {
+	var opt rpctest.Options
+	rpctest.RecreateConfig(&opt)
+
+	tn := rpctest.NewTendermint(cfg.App, &opt)
+
+	// rewriting the created config with test's one
+	tn.Config().Consensus = cfg.CoreCfg.Consensus
+
+	err := tn.Start()
+
+	return tn, err
```

Comment:
nit: you can directly `return tn, tn.Start()`

Reply:
Agreed. Done